utils/file.go

```diff
@@ -0,0 +1,122 @@
+package utils
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"os"
+	"io"
+	"syscall"
+	"fmt"
+	"errors"
+)
+
+// CopyDirectory copy all items from source to destination
+func CopyDirectory(scrDir, dest string) error {
+	entries, err := ioutil.ReadDir(scrDir)
+	if err != nil {
+			return err
+	}
+
+	if _, err := os.Stat(dest); os.IsNotExist(err) {
+		os.Mkdir(dest, 0755)
+	}
+
+	for _, entry := range entries {
+			sourcePath := filepath.Join(scrDir, entry.Name())
+			destPath := filepath.Join(dest, entry.Name())
+
+			fileInfo, err := os.Stat(sourcePath)
+			if err != nil {
+					return err
+			}
+
+			stat, ok := fileInfo.Sys().(*syscall.Stat_t)
+			if !ok {
+					return fmt.Errorf("failed to get raw syscall.Stat_t data for '%s'", sourcePath)
+			}
+
+			switch fileInfo.Mode() & os.ModeType{
+			case os.ModeDir:
+					if err := CreateIfNotExists(destPath, 0755); err != nil {
+							return err
+					}
+					if err := CopyDirectory(sourcePath, destPath); err != nil {
+							return err
+					}
+			case os.ModeSymlink:
+					if err := CopySymLink(sourcePath, destPath); err != nil {
+							return err
+					}
+			default:
+					if err := Copy(sourcePath, destPath); err != nil {
+							return err
+					}
+			}
+
+			if err := os.Lchown(destPath, int(stat.Uid), int(stat.Gid)); err != nil {
+					return err
+			}
+
+			isSymlink := entry.Mode()&os.ModeSymlink != 0
+			if !isSymlink {
+					if err := os.Chmod(destPath, entry.Mode()); err != nil {
+							return err
+					}
+			}
+	}
+	return nil
+}
+
+func Copy(srcFile, dstFile string) error {
+	if srcFile == dstFile {
+		return errors.New("source and direction are duplicated")
+	}
+
+	out, err := os.Create(dstFile)
+	if err != nil {
+		return err
+	}
+
+	defer out.Close()
+
+	in, err := os.Open(srcFile)
+	defer in.Close()
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Exists(filePath string) bool {
```

Comment:
Maybe 
`func Exist?(filePath string) bool` ?

Reply:
As I understand. It doesn't have a function? syntax in Golang.